x.ps1

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env pwsh
+
+# See x.sh for why these scripts exist.
+
+$xpy = Join-Path $PSScriptRoot x.py
+$xpy_args = @($xpy) + $args
+
+foreach ($python in "py", "python3", "python", "python2") {
+    # NOTE: this only tests that the command exists in PATH, not that it's actually
+    # executable. The latter is not possible in a portable way, see
+    # https://github.com/PowerShell/PowerShell/issues/12625.
```

Comment:
Hm, could we not simply try running the command? If `Start-Process` fails then we could simply move on to the next.

Reply:
> If Start-Process fails then we could simply move on to the next.
Does PowerShell distinguish "failed to start process" from "process ended with an error code"? The latter isn't correct because x.py will exit with an error if the build fails.

Reply:
Hm... I think it should throw an exception if the process failed to start. Which can then be caught and ignored. I would need to investigate more tho. Maybe something for a follow up PR?